unlock-app/src/components/interface/Error.js

```diff
@@ -6,52 +6,45 @@ import { resetError } from '../../actions/error'
 import Buttons from './buttons/layout'
 import ErrorMessage from '../helpers/ErrorMessage'
 
-export const Error = ({ error, close }) => {
-  if (!error) {
+export const Errors = ({ errors, close }) => {
+  const content = errors.map(error => (
+    <Wrapper key={error}>
+      <Error>{ErrorMessage(error)}</Error>
+      <SecondColumn cols={errors.length ? errors.length : 1}>
+        <Buttons.Close as="button" onClick={close} size="16px">
+          X
+        </Buttons.Close>
+      </SecondColumn>
+    </Wrapper>
+  ))
+  if (!content || !content.length) {
     return null
   }
 
-  let content = ErrorMessage(error)
-
-  return (
-    <Wrapper>
-      {content}
-      <Buttons.Close
-        as="button"
-        onClick={() => error && close(error)}
-        size="16px"
-      >
-        X
-      </Buttons.Close>
-    </Wrapper>
-  )
+  return <>{content}</>
 }
 
-const mapStateToProps = ({ errors }) => ({
-  // note: this is only showing the latest error. Soon we will rework
-  // the UI to support displaying multiple errors
-  error: errors.length ? errors[errors.length - 1] : null,
-})
+export const mapStateToProps = ({ errors }) => ({ errors })
 
 const mapDispatchToProps = dispatch => ({
-  close: error => {
```

Comment:
See above? Didn't we agree on the fact that each error would be closed independently?

Reply:
same